### PR TITLE
feat: add icons

### PR DESCRIPTION
**Pull request by scurker**
This became larger than I expected:

* Refactored `packages/react` to build with rollup
* Demo site now uses `packages/react/lib` instead of building from `src`
* SVG Icons
* Removed `typechecks` as it's no longer needed since we export types by default

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**
- [x] Follows the commit message policy, appropriate for next version
- [x] Tested for accessibility
- [x] Code is reviewed for security

### CODE REVIEW

File: docs/Demo/index.js [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import jsxStringify from 'react-element-to-jsx-string';
-import { Code } from '../../packages/react/src';
+import { Code } from '@deque/cauldron-react';
```
Comment (stephenmathieson) [inline docs/Demo/index.js:4]: 👍

File: packages/react/src/components/Icon/index.tsx [hunk 2]
```diff
@@ -8,24 +8,45 @@ export interface IconProps extends React.HTMLAttributes<HTMLDivElement> {
   type: string;
```
Comment (stephenmathieson) [inline packages/react/src/components/Icon/index.tsx:8]: I know it's a pain, but I _strongly_ suggest making `type` stricter here.

Something like:

```ts
type: 'Thing' | 'OtherThing' | 'etc...'
```

File: packages/react/src/components/Icon/index.tsx [hunk 3]
```diff
@@ -8,24 +8,45 @@ export interface IconProps extends React.HTMLAttributes<HTMLDivElement> {
...
-const Icon = ({ label, type, className, ...other }: IconProps) => {
+function Icon({ label, className, type, ...other }: IconProps) {
+  const [, name, direction] = type.match(/(.*)-(right|left|up|down)$/) || [
+    '',
+    type
+  ];
+  const [IconSVG, setIcon] = useState<React.ComponentType<any> | null>(null);
+
+  useEffect(() => {
+    import(`./icons/${name}.svg`)
```
Comment (stephenmathieson) [inline packages/react/src/components/Icon/index.tsx:19]: What's wrong with `import`ing all of the icons and tree-shaking what we don't use out?

Comment (scurker) [reply packages/react/src/components/Icon/index.tsx:19]: Dynamically importing means adding new icons is mostly drag and drop without having to modify the main `<Icon>` component. What's your concern here?

File: packages/react/src/components/Icon/index.tsx (hunk 2 above)
Comment (scurker) [reply packages/react/src/components/Icon/index.tsx]: I don't disagree, but we would need to make sure the type stay in sync with icons.

File: packages/react/src/components/Icon/index.tsx (hunk 3 above)
Comment (scurker) [reply packages/react/src/components/Icon/index.tsx:51]: Manually importing new icons and wrapping them with `<Icon>` just to import seems like unnecessary overhead when we can let rollup do the heavy lifting.

File: packages/react/src/components/Icon/index.tsx (hunk 2 above)
Comment (scurker) [reply packages/react/src/components/Icon/index.tsx:8]: Actually, typescript seems to be unhappy with dynamic strings when we define types.

```sh
src/index.ts → lib...
[!] (plugin typescript) Error: @rollup/plugin-typescript TS2322: Type 'string' is not assignable to type '"menu" | "code" | "arrow-up" | "arrow-down" | "arrow-left" | "arrow-right" | "bolt" | "checkbox-checked" | "checkbox-unchecked" | "chevron-up" | "chevron-down" | "chevron-left" | ... 21 more ... | "trash"'.
src/components/ExpandCollapsePanel/PanelTrigger.tsx (29:13)
29       <Icon type={`chevron-${open ? 'down' : 'right'}`} />
```

That's solvable by using `${open ? 'chevron-down' : 'chevron-right'}`, but I think for now I'd like to leave this as a non-breaking change.

Comment (stephenmathieson) [reply packages/react/src/components/Icon/index.tsx:8]: Ah bummer, OK. Thanks for trying this out.

File: packages/react/src/components/Icon/index.tsx (hunk 3 above)
Comment (stephenmathieson) [reply packages/react/src/components/Icon/index.tsx:19]: I understand that manually `import`ing all of them will be a pain, but I'm wondering how frequently we'll add/remove icon SVGs.

My main concern is dependent packages - are the consumers of `cauldron` setup to handle these `import`s?

File: packages/react/tsconfig.json [hunk 4]
```diff
@@ -1,7 +1,7 @@
 {
   "extends": "../../tsconfig.json",
   "compilerOptions": {
-    "outDir": "lib"
+    "typeRoots": ["node_modules/@types"]
```
Comment (stephenmathieson) [inline packages/react/tsconfig.json:4]: Isn't this the default? Why do we need to specify this?

File: packages/react/src/components/Icon/index.tsx (hunk 3 above)
Comment (scurker) [reply packages/react/src/components/Icon/index.tsx:19]: Yup! Rollup and dynamic-import-variables handles this for us, it outputs a map like this inside of `<Icon>`:

```js
function __variableDynamicImportRuntime0__(path) {
   switch (path) {
     case './icons/arrow.svg': return Promise.resolve().then(function () { return { 'default': require('./icons/arrow.svg.js') }; });
     case './icons/bolt.svg': return Promise.resolve().then(function () { return { 'default': require('./icons/bolt.svg.js') }; });
     case './icons/checkbox-checked.svg': return Promise.resolve().then(function () { return { 'default': require('./icons/checkbox-checked.svg.js') }; });
     /* ... and the rest of the icons */
     default: return Promise.reject(new Error("Unknown variable dynamic import: " + path));
   }
}

function Icon(_a) {
    React.useEffect(function () {
        __variableDynamicImportRuntime0__("./icons/" + name + ".svg")
            .then(function (icon) {
            setIcon(function () { return icon["default"]; });
        })["catch"](function (ex) {
            console.error("Could not find icon type \"" + type + "\".");
            setIcon(null);
        });
    }, [type])
  ...
}
```

As far as dependent packages are concerned, they don't even know there's a dynamic import. Assuming a bundler is used downstream it becomes an inline require.

File: packages/react/tsconfig.json (hunk 4 above)
Comment (scurker) [reply packages/react/tsconfig.json:4]: Good call, looks like it's not needed and we can safely remove it.

File: packages/react/src/components/Icon/index.tsx (hunk 3 above)
Comment (stephenmathieson) [reply packages/react/src/components/Icon/index.tsx:19]: Awesome!!

**Pull request merged**